File: week-1-c/mario/mario.c
#include <cs50.h>
#include <stdio.h>

void printf_repeat(char c, int times)
{
    while (--times >= 0)
    {
        printf("%c", c);
    }
}


int main(void)
{
    int height, width;
    char block = '#';
    char space = ' ';

    do
    {
        height = get_int("Height: ");
        width = height;
    }
    while (height < 1 || height > 8);

    for (int i = 1; i <= height; i++)
    {
        int num_blocks = i;
        int num_spaces = width - num_blocks;

        // left blocks
        printf_repeat(space, num_spaces);
        printf_repeat(block, num_blocks);

        printf("  ");

        // right bloacks
        printf_repeat(block, num_blocks);

        printf("\n");
    }

}

File: week-6-python/sentimental-mario-more/mario.py
from cs50 import get_int


def generate(height):
    for i in range(1, (height + 1)):
        spaces = ' ' * (height - i)
        blocks = '#' * i
        print(f'{spaces}{blocks}  {blocks}')


def get_height():
    while True:
        try:
            height = get_int("Height: ")
            if (height > 0 and height < 9):
                break
        except ValueError:
            None

    return height


if __name__ == "__main__":
    height = get_height()
    generate(height)